lib/fs.js

```diff
@@ -2479,3 +2479,452 @@ Object.defineProperty(fs, 'SyncWriteStream', {
   set: internalUtil.deprecate((val) => { SyncWriteStream = val; },
                               'fs.SyncWriteStream is deprecated.', 'DEP0061')
 });
+
+
+// The Promises API
+
+async function* readEverything(fd) {
+  let eof = false;
+  let position = 0;
+  const buffer = Buffer.allocUnsafe(kReadFileBufferLength);
+  while (!eof) {
+    const { length } =
+      await fs.promises.read(fd, buffer, 0, buffer.length, position);
+    if (length < buffer.length) {
+      eof = true;
+      return buffer.slice(0, length);  // Ended
+    } else {
+      position += length;
+      yield buffer.slice(0, length);
+    }
+  }
+}
+
+
+async function writeEverything(fd, buffer, offset, length, position) {
+  const { written } =
+    await fs.promises.write(fd, buffer, offset, length, position);
+  if (written !== length) {
+    offset += written;
+    length -= written;
+    if (position !== null)
+      position += written;
+    return writeEverything(fd, buffer, offset, length, position);
+  }
+}
+
+fs.promises = {
+  async access(path, mode = fs.F_OK) {
```

Comment:
Can you set `fs.access[util.promisify.custom] = fs.promises.access;` etc.?

Reply:
Yep. I will be